Lib/encodings/idna.py

```diff
@@ -168,11 +182,13 @@ def encode(self, input, errors='strict'):
         else:
             # ASCII name: fast path
             labels = result.split(b'.')
+            index = 0
             for label in labels[:-1]:
                 if not (0 < len(label) < 64):
-                    raise UnicodeError("label empty or too long")
+                    raise UnicodeEncodeError("idna", input, index, index+len(label), "label empty or too long")
+                index += len(label) + 1
             if len(labels[-1]) >= 64:
-                raise UnicodeError("label too long")
+                raise UnicodeEncodeError("idna", input, index, len(input), "label too long")
```

Comment:
```suggestion
                raise UnicodeEncodeError("idna", input, index, len(input),
                                         "label too long")
```

Reply:
Fixed the linebreaks of some of the longer exceptions to look like this